lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -517,6 +539,15 @@ static void buildModule(OpBuilder &builder, OperationState &result,
     portSyms.push_back(ports[i].sym ? ports[i].sym : builder.getStringAttr(""));
   }
 
+  if (llvm::all_of(portAnnotations, [](Attribute attr) {
+        return attr.cast<ArrayAttr>().empty();
+      }))
+    portAnnotations.clear();
+  if (llvm::all_of(portSyms, [](Attribute attr) {
+        return attr.cast<StringAttr>().getValue().empty();
+      }))
+    portSyms.clear();
+
```

Comment:
Would it be easier to do this same thing above instead of using the `addAnnos` boolean flag everywhere?  Also, this could use a comment describing why we clear out the array.

Reply:
Yeah that would definitely be an option. Probably much cleaner anyway!

Reply:
Implemented your approach @youngar, makes for much more readable code.